flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,442 @@
+        """
+
+        if not urls:
+            raise PluginError("Parameter 'urls' cannot be empty.")
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        data = {"urls": urls, "start": start}
+
+        if destination:
+            data["destination"] = destination
+        if tags:
+            data["tags"] = tags
+
+        try:
+            response = post(
+                f"{config['url']}/api/torrents/add-urls", json=data, cookies={"jwt": self.jwt}
+            )
+
+            if response.status_code == 200:
+                logger.debug(f"Successfully added {len(urls)} urls to Flood.")
+                return True
+
+            logger.debug(
+                f"Failed to add {len(urls)} urls to Flood. (status={response.status_code})"
+            )
+            return False
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to add torrent urls to Flood: {e}"
+            )
+
+    def add_torrent_files(
+        self,
+        config: dict,
+        files: list,
+        destination: str = None,
+        tags: list = None,
+        start: bool = True,
+    ) -> bool:
+        """
+        Add a list of torrent files to Flood.
+        Each file must be in base64 encoded format.
+        Will only attempt to add torrents if the JWT token is not expired or unset.
+        """
+
+        if not files:
+            raise PluginError("Parameter 'urls' cannot be empty.")
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        data = {"files": files, "start": start}
+
+        if destination:
+            data["destination"] = destination
+        if tags:
+            data["tags"] = tags
+
+        try:
+            response = post(
+                f"{config['url']}/api/torrents/add-files", json=data, cookies={"jwt": self.jwt}
+            )
+
+            if response.status_code == 200:
+                logger.debug(f"Successfully added {len(files)} files to Flood.")
+                return True
+
+            logger.debug(
+                f"Failed to add {len(files)} files to Flood. (status={response.status_code})"
+            )
+            return False
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to add torrent files to Flood: {e}"
+            )
+
+    def list_torrents(self, config: dict) -> dict:
+        """
+        List all torrents in Flood.
+        Will only attempt to list torrents if the JWT token is not expired or unset.
+        """
+
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        try:
+            response = get(f"{config['url']}/api/torrents", cookies={"jwt": self.jwt})
+
+            if response.status_code == 200:
```

Comment:
Same feedback as above

Reply:
Introduced new method in https://github.com/Winter/Flexget/commit/79fae7e for creating requests using the task requests session. It requires a list of successful status codes otherwise it will raise a PluginError halting the plugin.